toolchain/check/handle_function.cpp

```diff
@@ -303,8 +304,11 @@ static auto BuildFunctionDecl(Context& context,
         !function_info.param_patterns_id.is_valid() ||
         !context.inst_blocks().Get(function_info.param_patterns_id).empty() ||
         (return_type_id.is_valid() &&
-         return_type_id != context.GetInt32Type() &&
-         return_type_id != context.GetTupleType({}))) {
+         return_type_id != context.GetTupleType({}) &&
+         // TODO: Decide on valid return types for `i32`. Perhaps we should have
+         // an interface for this.
```

Comment:
```suggestion
         // TODO: Decide on valid return types for `Main.Run`. Perhaps we should have
         // an interface for this.
```

Feels like this meant `Run`